packages/components/src/core/Tag/index.testUtils.tsx

```diff
@@ -0,0 +1,285 @@
+import { CheckCircleOutline, WbSunny } from "@mui/icons-material";
+import { Args } from "@storybook/react";
+import React from "react";
+import Icon from "../Icon";
+import RawTag from "./index";
+import { ExtraTagProps } from "./style";
+
+const SDS_STYLES: ExtraTagProps["sdsStyle"][] = ["rounded", "square"];
+const SDS_TYPES: ExtraTagProps["sdsType"][] = ["primary", "secondary"];
+
+const DISPLAY_CONTENTS: React.CSSProperties = {
+  display: "contents",
+};
+
+const FONT_WEIGHT_NORMAL: React.CSSProperties = {
+  fontWeight: "normal",
+};
+
+const MID_LABEL: React.CSSProperties = {
+  ...FONT_WEIGHT_NORMAL,
+  borderStyle: "solid none none none",
+  gridColumn: "1 / 6",
+  justifySelf: "stretch",
+  paddingTop: 10,
+};
+
+export const AVAILABLE_ICON_OPTIONS = [
+  undefined,
+  <Icon sdsSize="l" sdsIcon="checkCircle" sdsType="button" />,
+  <Icon sdsSize="l" sdsIcon="loading" sdsType="button" />,
+  <WbSunny />,
+  <CheckCircleOutline />,
+];
+
+const iconOptions = AVAILABLE_ICON_OPTIONS.slice(0, 2);
+
+export const HOVER_OPTIONS = [true, false];
+
+export function CommonScreenshotTestDemo({
+  props,
+  colors,
+}: {
+  props: Args;
+  colors: ExtraTagProps["tagColor"][];
+}): JSX.Element {
+  // loop through all SDS_STYLES
+  return (
+    <>
+      {SDS_STYLES.map((sdsStyle) => {
+        return (
+          <PrimaryTagStyle
```

Comment:
should we rename `PrimaryTagStyle` to just `TagStyles`, since we are going through all `SDS_STYLES`?

Reply:
Yeah good point! I can make these updates after I merge this in!